### PR TITLE
Fix for #214 calls to TFAR functions in scheduled environment

### LINKED ISSUE #214: calls to TFAR functions in scheduled environment
As discussed in
- https://github.com/gruppe-adler/Shoot_and_Scoot.Tanoa/issues/44#issuecomment-2322932348
- https://github.com/gruppe-adler/Shoot_and_Scoot.Tanoa/issues/44#issuecomment-2329919267

`gruppe_adler_mod` calls to TFAR functions in [scheduled environment](https://community.bistudio.com/wiki/Scheduler#Scheduled_Environment). 

This is not supported by TFAR (as noted by @jokoho48 who in turn got the info from @dedmen).

We should wrap occurences of `call TFAR_fnc_*` in [CBA_fnc_directCall](https://cbateam.github.io/CBA_A3/docs/files/common/fnc_directCall-sqf.html).

If not there can be script erros. 
Such where noticed while testing with the [ICOM device](https://crowdedlight.github.io/Crows-Electronic-Warfare/spectrum/tracking/tfar-tracking.html?highlight=icom#listening-in) of CrowsEW.

### PR DESCRIPTION
**Pull request by AdlerSalbei**
fix #214

### CODE REVIEW

**Review by b-mayr-1984 — COMMENTED** [review]
@AdlerSalbei please let me know what you think of my suggestions for simplification.

File: addons/zeus/XEH_preClientInit.sqf [hunk 1]
```diff
@@ -38,77 +38,86 @@ if  (isClass (configFile >> "CfgPatches" >> "tfar_core")) then {
...
+                    _unit setVariable [QGVAR(channelSW), _channel, true];
+                    _unit setVariable [QGVAR(freqSW), _frequency, true];
                 };
-                _unit setVariable [QGVAR(channelAdditionalSW), _channel, true];
-                _unit setVariable [QGVAR(freqAdditionalSW), _frequency, true];
-            }else{
-                _unit setVariable [QGVAR(channelSW), _channel, true];
-                _unit setVariable [QGVAR(freqSW), _frequency, true];
-            };
+            }, _this] call CBA_fnc_directCall;
```
[the remark this answers is not in the document]
Comment (AdlerSalbei) [reply addons/zeus/XEH_preClientInit.sqf:59]: CBA_fnc_directCall returns a boolean, so i would need to handle the return diffrently, which would just make overcomplicate it.

File: addons/zeus/XEH_preClientInit.sqf [hunk 2]
```diff
@@ -38,77 +38,86 @@ if  (isClass (configFile >> "CfgPatches" >> "tfar_core")) then {
...
+                    _unit setVariable [QGVAR(channelLR), _channel, true];
+                    _unit setVariable [QGVAR(freqLR), _frequency, true];
                 };
-                _unit setVariable [QGVAR(channelAdditionalLR), _channel, true];
-                _unit setVariable [QGVAR(freqAdditionalLR), _frequency, true];
-            }else{
-                _unit setVariable [QGVAR(channelLR), _channel, true];
-                _unit setVariable [QGVAR(freqLR), _frequency, true];
-            };
+            }, _this] call CBA_fnc_directCall;
```
[the remark this answers is not in the document]
Comment (AdlerSalbei) [reply addons/zeus/XEH_preClientInit.sqf:86]: Same as with comment for "TFAR_event_OnSWchannelSet".

File: addons/zeus/XEH_preClientInit.sqf [hunk 3]
```diff
@@ -38,77 +38,86 @@ if  (isClass (configFile >> "CfgPatches" >> "tfar_core")) then {
...
+                            _unit setVariable [QGVAR(freqSW), _frequency, true];
+                        }else{
+                            if (_channel isEqualTo (_unit getVariable [QGVAR(channelAdditionalSW), -1])) then {
+                                _unit setVariable [QGVAR(freqAdditionalSW), _frequency, true];
+                            };
                         };
                     };
                 };
-            };
+            }, _this] call CBA_fnc_directCall;
```
[the remark this answers is not in the document]
Comment (AdlerSalbei) [reply addons/zeus/XEH_preClientInit.sqf:120]: Same as with comment for "TFAR_event_OnSWchannelSet".

File: addons/zeus/XEH_preClientInit.sqf (hunk 1 above)
[the remark this answers is not in the document]
Comment (b-mayr-1984) [reply addons/zeus/XEH_preClientInit.sqf:59]: No, I meant wrapping it like that:
```sqf
{
    private _frequency = [(call TFAR_fnc_activeSwRadio), (_channel +1)] call TFAR_fnc_getChannelFrequency;
} call CBA_fnc_directCall;
```

Then you don't have to worry about the return code I guess.
Or am I missing something?

[the remark this answers is not in the document]
Comment (b-mayr-1984) [reply addons/zeus/XEH_preClientInit.sqf:59]: > No, I meant wrapping it like that:
> 
> ```sqf
> {
>     private _frequency = [(call TFAR_fnc_activeSwRadio), (_channel +1)] call TFAR_fnc_getChannelFrequency;
> } call CBA_fnc_directCall;
> ```
> 
> Then you don't have to worry about the return code I guess. Or am I missing something?

Never mind this comment. I noticed my mistake here with regards to the `private` variable. 😑

But I think one could make the opening `{` of the direct call after `params`. That should work I guess.
I did this here: https://github.com/gruppe-adler/Shoot_and_Scoot.Tanoa/commit/4506aeb0859c681bdc3a707f3a9f33be53a51159?diff=unified&w=1

[the remark this answers is not in the document]
Comment (AdlerSalbei) [reply addons/zeus/XEH_preClientInit.sqf:59]: In your example that works, becouse there are no local variables. Param creates local variables that are not in scope if we do it after.

**Review by b-mayr-1984 — APPROVED** [review]
@AdlerSalbei thx for taking the time explaining the parameters thing to me. 🙂

**Comment by b-mayr-1984** [comment]
Tested using 

```
.\hemtt.exe build    
HEMTT 0.7.6
```

```
Original output filename: Arma3Retail_DX11_x64
Exe timestamp: 2024/03/21 14:45:00
Current time:  2024/09/11 21:59:02

Type: Public
Build: Stable
Version: 2.16.151618

Allocator: C:\Program Files (x86)\Steam\steamapps\common\Arma 3\Dll\tbb4malloc_bi_x64.dll [2017.0.0.0] [2017.0.0.0]
PhysMem: 16 GiB, VirtMem : 131072 GiB, AvailPhys : 7.2 GiB, AvailVirt : 131068 GiB, AvailPage : 15 GiB, PageSize : 4.0 KiB/2.0 MiB/HasLockMemory, CPUCount : 4
```

@AdlerSalbei unfortunately I get script errors when I change the TFAR short-range radio channel.

```
22:34:32 Error in expression <LrSettings;
};



((_settings param[2]) param [_channel,""])
>
22:34:32   Error position: <param [_channel,""])
>
22:34:32   Error Type Number,Not a Number, expected Number
22:34:32 File z\tfar\addons\core\functions\fnc_getChannelFrequency.sqf..., line 37
22:34:32  ➥ Context: 	[] L250 (/x/cba/addons/common/fnc_directCall.sqf)
	[] L250 (/x/cba/addons/common/fnc_directCall.sqf)
	[] L47 (x\GRAD\addons\zeus\XEH_preClientInit.sqf)
	[] L37 (z\tfar\addons\core\functions\fnc_getChannelFrequency.sqf)
 <- 
22:34:32 	[] L1 ()
	[] L1 ()
	[] L339 (/x/cba/addons/events/fnc_keyHandlerDown.sqf)
	[] L312 (/x/cba/addons/events/fnc_keyHandlerDown.sqf)
	[] L331 (/x/cba/addons/events/fnc_keyHandlerDown.sqf)
	[] L336 (/x/cba/addons/events/fnc_keyHandlerDown.sqf)
	[] L336 (/x/cba/addons/events/fnc_keyHandlerDown.sqf)
	[] L132 (z\tfar\addons\core\functions\fnc_initKeybinds.sqf)
	[] L25 (z\tfar\addons\core\functions\events\keys\fnc_processSWChannelKeys.sqf)
	[] L27 (z\tfar\addons\core\functions\events\keys\fnc_processSWChannelKeys.sqf)
	[] L30 (z\tfar\addons\core\functions\fnc_setSwChannel.sqf)
	[] L24 (z\tfar\addons\core\functions\events\handler\fnc_fireEventHandlers.sqf)
	[] L280 (/x/cba/addons/events/fnc_localEvent.sqf)
	[] L277 (/x/cba/addons/events/fnc_localEvent.sqf)
	[] L278 (/x/cba/addons/events/fnc_localEvent.sqf)
	[] L59 (x\GRAD\addons\zeus\XEH_preClientInit.sqf)
	[] L248 (/x/cba/addons/common/fnc_directCall.sqf)
```

**Comment by AdlerSalbei** [comment]
Thanks for the Test, I'll look have a look at it

**Comment by b-mayr-1984** [comment]
> Thanks for the Test, I'll look have a look at it

ccd63e1bc6a4ede6f3345280ae29dfc83b3fd5f0 does the trick.

Variables then track what I 
- set in the TFAR radio GUI
- select with the numpad

image

**Pull request merged**